### ISSUE TITLE
KubernetesClientBootstrapConfigurationTests tests failing with NPE on 2.0.x

### ISSUE THREAD

**Issue by spencergibb** [spring-cloud/spring-cloud-kubernetes#927]
Build started failing on Nov 10

https://jenkins.spring.io/view/Spring%20Cloud/view/CI/job/spring-cloud-kubernetes-2.0.x-ci/205/testReport/

org.springframework.cloud.kubernetes.client.config.KubernetesClientBootstrapConfigurationTests$KubernetesClientBootstrapConfigurationNotInsideK8s.bothMissing
 Error Details

Failed to load ApplicationContext

 Stack Trace
```
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'kubernetesKubectlCreateProcessor': Unsatisfied dependency expressed through field 'apiClient'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultApiClient' defined in class path resource [io/kubernetes/client/spring/extended/controller/config/KubernetesInformerAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.kubernetes.client.openapi.ApiClient]: Factory method 'defaultApiClient' threw exception; nested exception is java.lang.NullPointerException
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultApiClient' defined in class path resource [io/kubernetes/client/spring/extended/controller/config/KubernetesInformerAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.kubernetes.client.openapi.ApiClient]: Factory method 'defaultApiClient' threw exception; nested exception is java.lang.NullPointerException
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.kubernetes.client.openapi.ApiClient]: Factory method 'defaultApiClient' threw exception; nested exception is java.lang.NullPointerException
Caused by: java.lang.NullPointerException
```

org.springframework.cloud.kubernetes.client.config.KubernetesClientBootstrapConfigurationTests$KubernetesEnabledSecretsAndConfigDisabled.secretsOnlyPresent
 Error Details

Failed to load ApplicationContext

 Stack Trace
```
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'kubernetesKubectlCreateProcessor': Unsatisfied dependency expressed through field 'apiClient'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultApiClient' defined in class path resource [io/kubernetes/client/spring/extended/controller/config/KubernetesInformerAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.kubernetes.client.openapi.ApiClient]: Factory method 'defaultApiClient' threw exception; nested exception is java.lang.NullPointerException
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultApiClient' defined in class path resource [io/kubernetes/client/spring/extended/controller/config/KubernetesInformerAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.kubernetes.client.openapi.ApiClient]: Factory method 'defaultApiClient' threw exception; nested exception is java.lang.NullPointerException
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.kubernetes.client.openapi.ApiClient]: Factory method 'defaultApiClient' threw exception; nested exception is java.lang.NullPointerException
Caused by: java.lang.NullPointerException
```

**Comment by wind57** [comment]
to be honest that failing test is something I wrote, but I do not have access to that pipeline to tell what branch that is... I assume that is the something only available to the Spring team developers. If it could be re-produced here on github, I absolutely take a look.

**Comment by spencergibb** [comment]
Branch is 2.0.x

**Comment by ryanjbaxter** [comment]
Fixed via 38414c6.  Looks like we still need this PR in the 11.0.x branch https://github.com/kubernetes-client/java/pull/1963
